sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessor.java

```diff
@@ -229,14 +237,48 @@ private void receiveEvents(PartitionOwnership partitionOwnership) {
         PartitionProcessor partitionProcessor = this.partitionProcessorFactory
             .createPartitionProcessor(partitionContext, checkpointManager);
         partitionProcessor.initialize().subscribe();
-
+        final AtomicReference<Context> processSpanContext = new AtomicReference<>(Context.NONE);
         consumer.receive().subscribeOn(Schedulers.newElastic("PartitionPump"))
-            .subscribe(eventData -> partitionProcessor.processEvent(eventData).subscribe(unused -> {
-            }, partitionProcessor::processError),
+            .subscribe(eventData -> {
+                startScopedTracingSpan(eventData, processSpanContext);
+                partitionProcessor.processEvent(eventData).subscribe(unused -> {
+                }, partitionProcessor::processError);
+                endScopedTracingSpan(processSpanContext);
+            },
                 partitionProcessor::processError,
-                // Currently, there is no way to distinguish if the receiver was closed because
-                // another receiver with higher/same owner level(epoch) connected or because
-                // this event processor explicitly called close on this consumer.
+            // Currently, there is no way to distinguish if the receiver was closed because
```

Comment:
This can be undone? It looks like the comments are not at the same indentation level.